addons/account/models/account_move.py

```diff
@@ -1449,27 +1453,40 @@ def _compute_partner_credit_warning(self):
                            move.move_type == 'out_invoice' and \
                            move.company_id.account_use_credit_limit
             if show_warning:
+                current_amount = move.currency_id._convert(
+                    move.tax_totals['amount_total'],
+                    move.company_currency_id,
+                    move.company_id,
+                    move.date
+                )
```

Comment:
If the invoice is stored inside the database, you already have this value: `abs(amount_total_signed)`
If not, you must absolutely use the real total amount of your invoice! I hope you don't need this warning in edit mode because you can't get this information since the refactoring of 16.0.

Reply:
The warning needs to work before saving / while editing.
Else someone could confirm the invoice without ever seeing the warning (add some lines; press confirm).

I think the `tax_totals` field is the only field we can use to get the totals before saving?
And it should give us the correct total in document currency at least
(since it is used to display the tax and totals in the widget at the bottom)?

We just want to display a warning. So it is probably okay that the company currency amount is not 100% correct.
So just converting may be good enough?

Alternatively: I pushed a commit in which I changed the `tax_totals` to include the total amount in company currency
(we currently almost compute it already).
It is just an idea; could be bad?